### DISCUSSION TITLE
ethereum.decode does not recognise bool datatype

### DISCUSSION THREAD

**Discussion by 0xxlegolas** [graphprotocol/graph-tooling#1417] [Q&A]
Having this weird issue. Whenever there is bool type in the args ethereum.decode throws error : with message: `unexpected null `here is the code :
```
let data = Bytes.fromHexString("0x010000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000a")
let decoded = ethereum.decode("(bool,uint256,uint256)", data)!;
```
Expected values are : (true, 0, 10) 

You can try this with the below combination as well. 
```
let data = Bytes.fromHexString("0x01")
let decoded = ethereum.decode("(bool)", data)!; //this returns null
```
Any workaround for this ?

**Comment by azf20** [comment]
hey @0xxlegolas not sure what your exact use case is so may not help, but there are some examples on encoding and decoding booleans here: https://github.com/graphprotocol/graph-node/blob/6a7806cc465949ebb9e5b8269eeb763857797efc/tests/integration-tests/host-exports/src/mapping.ts#L72

**Comment by 0xxlegolas** [comment]
@azf20  Yes I have seen these examples. And I found out the issue. Apparently the bool value is `0x01` is not recognised by ethereum.encode. It only recognises `0x0000000000000000000000000000000000000000000000000000000000000001` as a bool where is bool is encoded as uint256 value. 

Below does not works 
```
let value = ethereum.decode("bool", Bytes.fromHexString("0x01"))

```
Below works 
```
let value = ethereum.decode("bool", Bytes.fromHexString("0x0000000000000000000000000000000000000000000000000000000000000001"))
```

**Comment by azf20** [comment]
OK great - closing as I think you have resolved, but reopen if you think a change in functionality is required here
